src/routes/hydra.ts

```diff
@@ -76,36 +75,33 @@ export default (
             kratosPublicEndpoint
               // We need to know who the user is for hydra
               .whoami(req as { headers: { [name: string]: string } })
-              .then( ({ body, response }) => {
+              .then( ({body, response} ) => {
                 // User is authenticated, accept the LoginRequest and tell Hydra
-                let acceptLoginRequest = new AcceptLoginRequest()
+                let acceptLoginRequest : AcceptLoginRequest = new AcceptLoginRequest()
+                // We should probably use "id" here but the gitlab-scenario only works via "email"
                 //acceptLoginRequest.subject = body.identity.id
-                acceptLoginRequest.subject = body.identity.traits.email
+                acceptLoginRequest.subject = (body.identity.traits as { email: string }).email || "fallback@mail.com"
```

Comment:
This is definitely something worth taking a note:
1. I couldn't get the gitlab-example to wotk without passing the mail (and not the ID. I assume that's the case because gitlab probably validates IDs as emails.
2. Other then that, we probably also need a better error-handling instead of a "fallback". This might also be a bit more complex as traits are not statically typed but we need to prevent warnings in order to "make docker" which fails otherwise.